query/src/main/java/org/infinispan/query/dsl/embedded/impl/SearchQueryMaker.java

```diff
@@ -327,11 +321,10 @@ public PredicateFinalStep visit(FullTextTermExpr fullTextTermExpr) {
    @Override
    public PredicateFinalStep visit(FullTextRegexpExpr fullTextRegexpExpr) {
       PropertyValueExpr propertyValueExpr = (PropertyValueExpr) fullTextRegexpExpr.getChild();
-      String regexp = fullTextRegexpExpr.getRegexp();
 
-      // TODO TODO HSEARCH-3884 Support regexp queries
-      RegexpQuery nativeQuery = new RegexpQuery(new Term(propertyValueExpr.getPropertyPath().asStringPath(), regexp));
-      return predicateFactory.fromLuceneQuery(nativeQuery);
+      return predicateFactory.regexp()
+            .field(propertyValueExpr.getPropertyPath().asStringPath())
+            .matching(fullTextRegexpExpr.getRegexp());
```

Comment:
Be aware that you won't get the same behavior as before, because we disable all flags by default, while `new RegexpQuery` enables all flags by default.

If you want backwards-compatible behavior, you'll have to add `.flags( RegexpQueryFlag.COMPLEMENT, RegexpQueryFlag.INTERVAL,  RegexpQueryFlag.INTERSECTION,  RegexpQueryFlag.ANY_STRING )` here.

Otherwise, I'd recommend documenting the behavior change or making flags configurable.

Reply:
Actually we don't have enough tests to cover all these sub-cases, right let's keep the behaviour consistent.